File: adventofcode/cpp/year2019/day24/day24_test.cc
#include "adventofcode/cpp/year2019/day24/day24.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day24, Part1Example1) {
  std::ifstream input("adventofcode/cpp/year2019/day24/testdata/ex");
  std::string output = "2129920";
  adventofcode::answer_t a = day24::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day24, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day24/actual.in");
  std::string output = "28717468";
  adventofcode::answer_t a = day24::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day24, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day24/actual.in");
  std::string output = "2014";
  adventofcode::answer_t a = day24::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

File: adventofcode/java/year2016/day24/Day24Test.java
package adventofcode.java.year2016.day24;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day24Test {
    @Test
    public void part1Example() throws IOException {
        try (var input = new FileReader("adventofcode/java/testdata/year2016/day24/example")) {
            var output = "14";
            var result = Day24.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day24/actual.in")) {
            var output = "470";
            var result = Day24.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day24/actual.in")) {
             var output = "720";
             var result = Day24.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}
